presidio-anonymizer/presidio_anonymizer/operators/operators_factory.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from typing import Dict
+
+from presidio_anonymizer.entities import InvalidParamException
+from presidio_anonymizer.operators import OperatorType, Operator
+
+
+class OperatorsFactory:
+    """Operators factory to get the correct operator class."""
+
+    _anonymizers: Dict = None
+    _deanonymizers: Dict = None
+
+    def __init__(self):
+        self.logger = logging.getLogger("presidio-anonymizer")
+
+    def create_operator_class(self, operator_name: str,
+                              operator_type: OperatorType) -> Operator:
+        """
+        Extract the operator class from the operators list.
+
+        :param operator_type: Either Anonymize or Decrypt to defer between operators.
+        :type operator_name: operator name.
+        :return: operator class entity.
+        """
+        operator_class = {
+            OperatorType.Anonymize: OperatorsFactory.get_anonymizers().get(
+                operator_name),
+            OperatorType.Deanonymize: OperatorsFactory.get_deanonymizers().get(
+                operator_name),
+        }.get(operator_type)
```

Comment:
__get_operators_by_type can be used along with the operator_type instead of constructing a dict, or even better this dict can be evaluated once for all calls

Reply:
Done.